lib/matplotlib/colorbar.py

```diff
@@ -900,6 +900,409 @@ def remove(self):
+
+    def draw_all(self):
+        normx = self.norm.norm1
+        normy = self.norm.norm2
+        self._xvalues, self._xboundaries = self._process_values(norm=normx)
+        self._yvalues, self._yboundaries = self._process_values(norm=normy)
+        X, Y = self._mesh()
+        CX, CY = np.meshgrid(self._xvalues, self._yvalues)
+        self.update_ticks()
+        if self.filled:
+            self._add_solids(X, Y, [CX, CY])
+
+    def config_axis(self):
+        ax = self.ax
+        ax.set_navigate(False)
+
+        ax.yaxis.set_label_position('right')
+        ax.yaxis.set_ticks_position('right')
+
+        ax.xaxis.set_label_position('bottom')
+        ax.xaxis.set_ticks_position('bottom')
+
+        self._set_label()
+
+    def update_ticks(self):
+        """
+        Force the update of the ticks and ticklabels. This must be
+        called whenever the tick locator and/or tick formatter changes.
+        """
+        ax = self.ax
+        xticks, xticklabels, xoffset_string = self._ticker(self.norm.norm1)
+        yticks, yticklabels, yoffset_string = self._ticker(self.norm.norm2)
+
+        ax.xaxis.set_ticks(xticks)
+        ax.set_xticklabels(xticklabels)
+        ax.xaxis.get_major_formatter().set_offset_string(xoffset_string)
+
+        ax.yaxis.set_ticks(yticks)
+        ax.set_yticklabels(yticklabels)
+        ax.yaxis.get_major_formatter().set_offset_string(yoffset_string)
+
+    def set_ticks(self, xticks, yticks, update_ticks=True):
+        if cbook.iterable(xticks):
+            self.xlocator = ticker.FixedLocator(xticks, nbins=len(xticks))
+        else:
+            self.xlocator = xticks
+
+        if cbook.iterable(yticks):
+            self.ylocator = ticker.FixedLocator(yticks, nbins=len(yticks))
+        else:
+            self.ylocator = yticks
+
+        if update_ticks:
+            self.update_ticks()
+        self.stale = True
+
+    def set_ticklabels(self, xticklabels=None, yticklabels=None,
+                       update_ticks=True):
+        """
+        set tick labels. Tick labels are updated immediately unless
+        update_ticks is *False*. To manually update the ticks, call
+        *update_ticks* method explicitly.
+        """
+        if xticklabels is not None or yticklabels is not None:
+            if isinstance(self.xlocator, ticker.FixedLocator):
+                self.xformatter = ticker.FixedFormatter(xticklabels)
+
+            if isinstance(self.ylocator, ticker.FixedLocator):
+                self.yformatter = ticker.FixedFormatter(yticklabels)
+
+            if update_ticks:
+                self.update_ticks()
+
+            self.stale = True
+            return
+
+        warnings.warn("set_ticks() must have been called.")
+        self.stale = True
+
+    def _set_label(self):
+        self.ax.set_ylabel(self._ylabel, **self._labelkw)
+        self.ax.set_xlabel(self._xlabel, **self._labelkw)
+        self.stale = True
+
+    def set_label(self, xlabel, ylabel, **kw):
```

Comment:
The different signature from the `ColorBar` base class is a bit problematic, but not sure what to do about it.